iambic/plugins/v0_1_0/aws/organizations/scp/models.py

```diff
@@ -0,0 +1,582 @@
+                    list(chain.from_iterable(changes_made))
+                )
+
+            # name and identifier must match
+            self.identifier = current_policy.get("Name", self.identifier)
+            self.write()
+
+        self.__log_after_apply(account_change_details, log_params)
+
+        return account_change_details
+
+    def is_delete_action(self, aws_account):
+        deleted = self.get_attribute_val_for_account(aws_account, "deleted", False)
+        if isinstance(deleted, list):
+            deleted = deleted[0].deleted
+
+        return deleted
+
+    def __log_after_apply(self, account_change_details, log_params):
+        if ctx.execute and not account_change_details.exceptions_seen:
+            log.debug(
+                "Successfully finished execution on account for resource",
+                changes_made=bool(account_change_details.proposed_changes),
+                **log_params,
+            )
+        elif account_change_details.exceptions_seen:
+            log.error(
+                "Unable to finish execution on account for resource",
+                exceptions_seen=[
+                    cd.exceptions_seen for cd in account_change_details.exceptions_seen
+                ],
+                **log_params,
+            )
+        else:
+            log.debug(
+                "Successfully finished scanning for drift on account for resource",
+                requires_changes=bool(account_change_details.proposed_changes),
+                **log_params,
+            )
+
+    @staticmethod
+    def factory_template_props(
+        account_id: str,
+        policy: ServiceControlPolicyItem,
+        config: AWSConfig,
+        organization: AWSOrganization,
+    ):
+        template_params = dict(
+            identifier=policy.Name,
+            account_id=account_id,
+            org_id=organization.org_id,
+        )
+
+        template_properties = dict(
+            policy_id=policy.Id,
+            policy_name=policy.Name,
+            description=policy.Description,
+            type=policy.Type,
+            aws_managed=policy.AwsManaged,
+            policy_document=dict(
+                version=policy.PolicyDocument.Version,
+                statement=policy.PolicyDocument.Statement,
+            ),
+        )
+
+        if policy.Targets:
+            template_properties.update(
+                targets=PolicyTargetProperties.parse_obj(
+                    PolicyTargetProperties.parse_targets(
+                        [normalize_dict_keys(t.dict()) for t in policy.Targets],
+                        config,
+                    )
+                )  # type: ignore
+            )
+
+        if policy.Tags:
+            template_properties.update(
+                tags=[normalize_dict_keys(t.dict()) for t in policy.Tags]  # type: ignore
+            )
+        return template_params, template_properties
+
+
+class PolicyStatementItem(OutputModels):
+    Sid: Optional[str]
```

Comment:
I think this already is defined here: https://github.com/noqdev/iambic/blob/508af0bd3665e65e7a300f463794a70e51b1ef0a/iambic/plugins/v0_1_0/aws/iam/policy/models.py#L140

Reply:
the `OutputModels` are just responses from boto3, sometimes are equal but not always. Also, these responses doesn't need AccessModel and ExpiryModel.

We could use these response models as mixins? like: `PolicyStatement(PolicyStatementMixin, AccessModel, ExpiryModel):`

What did I difference between these models and other ones? There are cases where the format is not the same. e.g.: the template properties. In our case, we have "targets -> org_units, accounts" and the response from boto3 doesn't have that structure.

Reply:
Got it, if it's for the response message, probably not a big deal in that case. Carry on!